sky/clouds/cudo.py

```diff
@@ -0,0 +1,354 @@
+
+        project_id, error = cudo_api.get_project_id()
+        if error is not None:
+            return False, (
+                f'Error getting project '
+                f'{common_utils.format_exception(error, use_bracket=True)}')
+        try:
+            api = cudo_api.virtual_machines()
+            api.list_vms(project_id)
+            return True, None
+        except ApiException as e:
+            return False, (
+                f'Error calling API '
+                f'{common_utils.format_exception(e, use_bracket=True)}')
+
+    def get_credential_file_mounts(self) -> Dict[str, str]:
+        return {
+            f'~/.config/cudo/{filename}': f'~/.config/cudo/{filename}'
+            for filename in _CREDENTIAL_FILES
+        }
+
+    @classmethod
+    def get_current_user_identity(cls) -> Optional[List[str]]:
+        # NOTE: used for very advanced SkyPilot functionality
+        # Can implement later if desired
+        return None
+
+    def instance_type_exists(self, instance_type: str) -> bool:
+        return service_catalog.instance_type_exists(instance_type, 'cudo')
+
+    def validate_region_zone(self, region: Optional[str], zone: Optional[str]):
+        return service_catalog.validate_region_zone(region, zone, clouds='cudo')
+
+    def accelerator_in_region_or_zone(self,
+                                      accelerator: str,
+                                      acc_count: int,
+                                      region: Optional[str] = None,
+                                      zone: Optional[str] = None) -> bool:
+        return service_catalog.accelerator_in_region_or_zone(
+            accelerator, acc_count, region, zone, 'cudo')
+
+    @classmethod
+    def query_status(cls, name: str, tag_filters: Dict[str, str],
+                     region: Optional[str], zone: Optional[str],
+                     **kwargs) -> List[status_lib.ClusterStatus]:
+        del tag_filters, region, zone, kwargs  # Unused.
+
+        # pylint: disable=import-outside-toplevel,unused-import
+        import sky.provision.cudo.cudo_wrapper as cudo_wrapper
+        status_map = {
+            'init': status_lib.ClusterStatus.INIT,
+            'pend': status_lib.ClusterStatus.INIT,
+            'prol': status_lib.ClusterStatus.INIT,
+            'boot': status_lib.ClusterStatus.INIT,
+            'runn': status_lib.ClusterStatus.UP,
+            'stop': status_lib.ClusterStatus.STOPPED,
+            'susp': status_lib.ClusterStatus.STOPPED,
+            'done': status_lib.ClusterStatus.STOPPED,
+            'poff': status_lib.ClusterStatus.STOPPED,
+        }
+        status_list = []
+        vms = cudo_wrapper.list_instances()
+        for node in vms:
+            if vms[node]['name'] == name:
+                node_status = status_map[node['status']]
+                if node_status is not None:
+                    status_list.append(node_status)
+        return status_list
```

Comment:
Seems this is already implemented in `sky/provision/cudo/instance.py`, it should be fine to be removed.
```suggestion
```